File: 1715b_ba.cpp
/**
 * @file 1715b_ba.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Beautiful Array(https://codeforces.com/problemset/problem/1715/B)
 * @version 0.1
 * @date 2024-05-13
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	short t;
	int n, k, b, i;
	long long s;
	vector<long long> a;

	cin >> t;
	while (t--)
	{
		cin >> n >> k >> b >> s;
		if (1ll * (k - 1) * (n - 1) < (s - ((long long)(1ll * k * b) + k - 1)) || s < (long long)(1ll * k * b))
		{
			cout << "-1\n";
			continue;
		}
		a.resize(n);
		a.assign(n, 0);
		a[0] = min(s, (long long)k * b + k - 1);
		s -= a[0];
		i = 1;
		while (s)
		{
			a[i++] = min(s, (long long)k - 1);
			s -= min((long long)k - 1, s);
		}
		for (long long ai : a)
			cout << ai << ' ';
		cout << '\n';
	}
	return (EXIT_SUCCESS);
}

File: 1715b_ba.py
"""
@file: 1715b_ba.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Beautiful Array(https://codeforces.com/contest/1715/problem/B)
@version: 1.0
@date: 2024-11-19

@copyright: Copyright (c) 2024
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    t = int(input())
    for _ in range(t):
        n, k, b, s = [int(x) for x in input().split()]
        if k * b > s or (k - 1) * (n - 1) < s - k * (b + 1) + 1:
            print(-1)
            continue
        res = [0] * n
        res[0], s = min(k * (b + 1) - 1, s), s - min(k * (b + 1) - 1, s)
        for i in range(1, n):
            res[i], s = min(s, k - 1), s - min(s, k - 1)
        print(" ".join(str(x) for x in res))


if __name__ == "__main__":
    main()
